datafusion/common/src/scalar/mod.rs

```diff
@@ -974,6 +974,129 @@ impl ScalarValue {
         )
     }
 
+    /// Create a Null instance of ScalarValue for this datatype
+    ///
+    /// Example
+    /// ```
+    /// use datafusion_common::ScalarValue;
+    /// use arrow::datatypes::DataType;
+    ///
+    /// let scalar = ScalarValue::try_new_null(&DataType::Int32).unwrap();
+    /// assert_eq!(scalar.is_null(), true);
+    /// assert_eq!(scalar.data_type(), DataType::Int32);
+    /// ```
+    pub fn try_new_null(data_type: &DataType) -> Result<Self> {
```

Comment:
Thank you @cj-zhukov  🙏 

I think this duplicates much of the code in the `TryFrom` impl -- can you please change `TryFrom` to call this new function to avoid the duplication?